File: app/src/main/java/com/example/museumapplication/utils/NetCheckUtils.java
package com.example.museumapplication.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.util.Log;

public class NetCheckUtils {
    private static final String TAG = "NetCheckUtils";

    private NetCheckUtils() {}

    /**
     * Check Is Network Available
     *
     * @param context Context
     * @return true:Network is available
     */
    public static boolean isNetworkAvailable(Context context) {
        boolean mobileConnection = isMobileConnection(context);
        boolean wifiConnection = isWifiConnection(context);
        if (!mobileConnection && !wifiConnection) {
            Log.i(TAG, "No network available");
            return false;
        }
        return true;
    }

    /**
     * Is Mobile Connection
     *
     * @param context Context
     * @return true:Mobile is connection
     */
    public static boolean isMobileConnection(Context context) {
        Object object = context.getSystemService(Context.CONNECTIVITY_SERVICE);
        if (!(object instanceof ConnectivityManager)) {
            return false;
        }
        ConnectivityManager manager = (ConnectivityManager) object;
        NetworkInfo networkInfo = manager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE);

        if (networkInfo != null && networkInfo.isConnected()) {
            return true;
        }
        return false;
    }

    /**
     * Is WIFI Connection
     *
     * @param context Context
     * @return true:wifi is connection
     */
    public static boolean isWifiConnection(Context context) {
        Object object = context.getSystemService(Context.CONNECTIVITY_SERVICE);
        if (!(object instanceof ConnectivityManager)) {
            return false;
        }
        ConnectivityManager manager = (ConnectivityManager) object;
        NetworkInfo networkInfo = manager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);

        if (networkInfo != null && networkInfo.isConnected()) {
            return true;
        }
        return false;
    }
}

File: app/src/main/java/com/example/museumapplication/utils/virtualGuide/NetCheckUtils.kt
package com.example.museumapplication.utils.virtualGuide

import android.content.Context
import android.net.ConnectivityManager
import android.util.Log

object NetCheckUtils {
    private const val TAG = "NetCheckUtils"

    /**
     * Check Is Network Available
     *
     * @param context Context
     * @return true:Network is available
     */
    fun isNetworkAvailable(context: Context): Boolean {
        val mobileConnection = isMobileConnection(context)
        val wifiConnection = isWifiConnection(context)
        if (!mobileConnection && !wifiConnection) {
            Log.i(TAG, "No network available")
            return false
        }
        return true
    }

    /**
     * Is Mobile Connection
     *
     * @param context Context
     * @return true:Mobile is connection
     */
    private fun isMobileConnection(context: Context): Boolean {
        val `object` = context.getSystemService(Context.CONNECTIVITY_SERVICE) as? ConnectivityManager
                ?: return false
        val networkInfo = `object`.getNetworkInfo(ConnectivityManager.TYPE_MOBILE)
        return networkInfo != null && networkInfo.isConnected
    }

    /**
     * Is WIFI Connection
     *
     * @param context Context
     * @return true:wifi is connection
     */
    private fun isWifiConnection(context: Context): Boolean {
        val `object` = context.getSystemService(Context.CONNECTIVITY_SERVICE) as? ConnectivityManager
                ?: return false
        val networkInfo = `object`.getNetworkInfo(ConnectivityManager.TYPE_WIFI)
        return networkInfo != null && networkInfo.isConnected
    }
}